lxd/containers_post.go

```diff
@@ -528,14 +532,89 @@ func createFromCopy(d *Daemon, req *api.ContainersPost) Response {
 	return OperationResponse(op)
 }
 
+func createFromBackup(d *Daemon, data []byte) Response {
+	if len(data) == 0 {
+		return BadRequest(fmt.Errorf("must specify backup source"))
+	}
+
+	bInfo, err := getBackupInfo(bytes.NewReader(data))
+	if err != nil {
+		return BadRequest(err)
+	}
+
+	run := func(op *operation) error {
+		// Dump tarball to storage
+		err = containerCreateFromBackup(d.State(), *bInfo, data)
+		if err != nil {
+			return err
+		}
+
+		body, err := json.Marshal(&internalImportPost{
+			Name:  bInfo.Name,
+			Force: true,
+		})
+		if err != nil {
+			return err
+		}
+
+		resp := internalImport(d, &http.Request{
+			Body: ioutil.NopCloser(bytes.NewReader(body)),
+		})
+
+		if resp.String() != "success" {
+			return errors.New(resp.String())
+		}
+
+		// Unmount volumes
+		c, err := containerLoadByName(d.State(), bInfo.Name)
```

Comment:
That comment should either move down to the next function call or go away. :)

Reply:
fixed